yt_dlp/extractor/hrfensehen.py

```diff
@@ -72,21 +74,30 @@ def _real_extract(self, url):
         description = self._html_search_meta(
             ['description'], webpage)
 
-        loader_str = unescapeHTML(self._search_regex(r"data-new-hr-mediaplayer-loader='([^']*)'", webpage, "ardloader"))
+        loader_str = unescapeHTML(self._search_regex(r"data-(?:new-)?hr-mediaplayer-loader='([^']*)'", webpage, "ardloader"))
         loader_data = json.loads(loader_str)
 
         info = {
             'id': video_id,
             'title': title,
             'description': description,
             'formats': self.extract_formats(loader_data),
-            'timestamp': self.extract_airdate(loader_data)
+            'duration': self.extract_duration(loader_data),
+            'timestamp': self.extract_datetime(webpage)
         }
 
-        if "subtitle" in loader_data:
-            info["subtitles"] = {"de": [{"url": loader_data["subtitle"]}]}
+        if "subTitles" in loader_data['mediaCollection']:
+            subt = traverse_obj(loader_data, ('mediaCollection', 'subTitles'))
+            if subt is not None:
+                subtitle = subt[0]['sources'][0]['url']
+                info["subtitles"] = {"de": [{"url": subtitle}]}
```

Comment:
```suggestion
        subtitle = traverse_obj(loader_data, ('mediaCollection', 'subTitles', 0, 'sources', 0, 'url'))
        if subtitle:
            info['subtitles'] = {'de': [{'url': subtitle}]}
```
This can be further simplified by moving it above `info = {...}`

Then u can also move `thumbnails` above it, and directly `return {...}`